### PR TITLE
Enables production data from CSV file and fixes failing probability distribution test

### PR DESCRIPTION
**Pull request by edubarrosTNO**
*Enables production data to be loaded from CSV file and fixes failing probability distribution test*

---

### Contributor checklist

- [x] :tada: This PR **partially addresses** #382.
- [x] :scroll: I have broken down my PR into the following tasks:
   - [x] Modify parser to allow for new optional entry in the configuration file.
   - [x] Create class `CSVData` as an alternative to `FlowData`.
   - [x] Replace production data dataframe loaded from reference simulation by data loaded from CSV file.
   - [x] Fix probability distributions test function (deprecated replacement of NaNs by None)
   - [x] Update documentation and `CHANGELOG.md`.
   - [x] Change `config_parser` for new position of observation `vectors` entry in configuration file.
- [x] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [x] :book: I have considered adding a new entry in `CHANGELOG.md`.
- [x] :books: I have considered updating the documentation.

### CODE REVIEW

File: src/flownet/config_parser/_config_parser.py [hunk 1]
```diff
@@ -69,6 +69,17 @@ def _to_abs_path(path: Optional[str]) -> str:
                     "data_source": {
                         MK.Type: types.NamedDict,
                         MK.Content: {
+                            "database": {
```
[the remark this answers is not in the document]
Comment (edubarrosTNO) [reply src/flownet/config_parser/_config_parser.py:72]: 👍 , done!

File: src/flownet/data/from_csv.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        super().__init__()
+
+        self._input_data: Path = Path(input_data)
+
+    # pylint: disable=too-many-branches
+    def _production_data(self) -> pd.DataFrame:
+        """
+        Function to read production data for all producers and injectors from a CSV file.
+
+        Returns:
```
[the remark this answers is not in the document]
Comment (edubarrosTNO) [reply src/flownet/data/from_csv.py:29]: I don't know if I understand the problem or how to solve it. Are you saying that we should check if all supported summary vectors are present in the CSV file? If so, is there a list of all supported summary vectors anywhere in the code?

**Comment by edubarrosTNO** [comment]
Previous CI test failed because configuration files from https://github.com/equinor/flownet-testdata must be updated (new position of observation vectors entry, PR https://github.com/equinor/flownet-testdata/pull/52). Testdata branch has been changed to verify CI before merging

**Comment by edubarrosTNO** [comment]
> Previous CI test failed because configuration files from https://github.com/equinor/flownet-testdata must be updated (new position of observation vectors entry, PR [equinor/flownet-testdata#52](https://github.com/equinor/flownet-testdata/pull/52)). Testdata branch has been changed to verify CI before merging

All CI tests have passed when using the branch of https://github.com/equinor/flownet-testdata/pull/52 - that PR can be merged safely. Next commit of this PR #425 reverts CI back to equinor/master branch, see below

**Review by olelod — APPROVED** [review]
LGTM

**Pull request merged**